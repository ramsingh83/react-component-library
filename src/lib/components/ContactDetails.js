import React from "react";
import "./ContactDetails.css";

const ContactDetails = ({
  label,
  placeholder,
  email,
  handleOnInputChanged,
  invalid,
  required
}) => (
  <div className="form-item">
    <label htmlFor="email-id">
      <span className={`input-label ${required ? 'element-required' : ''}`}>{label}</span>
      <input
        id="email-id"
        type="text"
        className={`form-input ${invalid ? 'invalid' : ''}`}
        aria-describedby="email-error"
        aria-label={label}
        placeholder={placeholder}
        onChange={handleOnInputChanged}
        value={email}
        maxLength="128"
        aria-required="true"
        aria-invalid={!!invalid} />
    </label>
    <div id="email-error" className="error-ifo"></div>
  </div>
);

export default ContactDetails;
