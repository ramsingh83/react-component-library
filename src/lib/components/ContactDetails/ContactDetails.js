import React from 'react';

const ContactDetails = ({
  label,
  placeholder,
  email,
  handleOnInputChanged,
  invalid,
  required
}) => (
  <div className="form-item">
    <label htmlFor="email-id">
      <span className="input-label">{label}</span>
      {required ? <span style={{ color: 'red' }}>*</span> : ''}
      <input
        id="email-id"
        type="text"
        className={`form-input ${invalid ? 'invalid' : ''}`}
        aria-describedby="email-error"
        aria-label={label}
        placeholder={placeholder}
        onChange={handleOnInputChanged}
        value={email}
        maxLength="128"
        aria-required="true"
        aria-invalid={!!invalid} />
    </label>
    <div id="email-error" className="error-ifo"></div>
    {console.log('test')}
  </div>
);

export default ContactDetails;
