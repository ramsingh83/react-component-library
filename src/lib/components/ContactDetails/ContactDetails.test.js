import React from 'react';
import renderer from 'react-test-renderer';
import ContactDetails from './ContactDetails';

describe('ContactDetails', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<ContactDetails label="Email" placeholder="name@example.com" email="test@eamil.com" handleOnInputChanged={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
