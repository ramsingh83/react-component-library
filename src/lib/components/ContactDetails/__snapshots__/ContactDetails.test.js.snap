// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetails renders properly 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="email-id"
  >
    <span
      className="input-label"
    >
      Email
    </span>
    
    <input
      aria-describedby="email-error"
      aria-invalid={false}
      aria-label="Email"
      aria-required="true"
      className="form-input "
      id="email-id"
      maxLength="128"
      onChange={[Function]}
      placeholder="name@example.com"
      type="text"
      value="test@eamil.com"
    />
  </label>
  <div
    className="error-ifo"
    id="email-error"
  />
</div>
`;
